### ISSUE TITLE
Page hierarchy explorer doesn't always select

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#2360]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1. build Toolpad `yarn && yarn release:build`
2. start Toolpad dev on the basic backend fixture: `yarn toolpad dev test/integration/backend-basic/fixture`
3. open the "basic" page
4. try selecting items in the page hierarchy explorer

### Current behavior 😯

1. It doesn't always select the item (see video)
2. hovering over page rows doesn't highlight them
3. the title "components" is very ambiguous. Components for me are like templates, they are blueprints. When they're instantiated on the page I prefer to talk about nodes, or elements.

https://github.com/mui/mui-toolpad/assets/2109932/89208dd8-5a26-43ec-9927-36651b9b380a

### Expected behavior 🤔

1. It always selects the node when I click it
2. hovering over page rows highlights the node in the canvas
3. I propose we change the title to "page hierarchy"

**Comment by bharatkashyap** [comment]
@Janpot Is there any additional information you could share? Context/Browser/error messages in the console? I can't replicate 1. without those.


https://github.com/mui/mui-toolpad/assets/19550456/2b9e58fb-2d79-430f-b40b-7239b293c1b3

Agree, I'll make changes for 2. and 3.

**Comment by Janpot** [comment]
Yes, it seems to happen when I click the text itself, or the vertical center in the whitespace left of the text, basically anything in this area:

![Screenshot 2023-07-25 at 10 40 50](https://github.com/mui/mui-toolpad/assets/2109932/5245ce29-4ede-432e-b1d3-89398e34e501)

**Comment by Janpot** [comment]
I also updated the issue with 4. and 5. that I ran into.

**Comment by bharatkashyap** [comment]
On 3., "Components" makes sense to me given that :

- We call the catalog which contains them the "Component Library" 

- Benchmarking:

| Appsmith | Dynaboard | Retool |
| -------| -----------|  --------- |
| <img width="256" alt="Screenshot 2023-07-31 at 4 42 11 PM" src="https://github.com/mui/mui-toolpad/assets/19550456/be419ebe-de37-40b8-bcaf-c39252e7da05"> | <img width="259" alt="Screenshot 2023-07-31 at 4 36 17 PM" src="https://github.com/mui/mui-toolpad/assets/19550456/531ba86e-47d0-4f56-b421-3d8dd096ea46"> | <img width="229" alt="Screenshot 2023-07-31 at 4 34 13 PM" src="https://github.com/mui/mui-toolpad/assets/19550456/6020a00d-aeb3-4d52-be0a-f482a3d429f1"> |

The title of the section that shows the tree is the same as the entity which you drag on to the page - So "add new widget" translates to a "Widgets" explorer; similar for Retool and Dynaboard

- Adding a new term for a component when instantiated on the page seems like an unnecessary complication

**Comment by Janpot** [comment]
> * Adding a new term for a component when instantiated on the page seems like an unnecessary complication

The distinction between "template" and "instance" is a very natural one for programmers:

- a class vs. an object
- a React component vs. a jsx element
-

**Comment by bharatkashyap** [comment]
> > * Adding a new term for a component when instantiated on the page seems like an unnecessary complication
> 
> The distinction between "template" and "instance" is a very natural one for programmers:
> 
> * a class vs. an object
> * a React component vs. a jsx element

- Updated the title to say "Page Hierarchy"

**Comment by prakhargupta1** [comment]
A code component currently comes without any icon, let's use the icon that we are using in the Component library.
<img width="247" alt="Screenshot 2023-08-07 at 2 00 57 PM" src="https://github.com/mui/mui-toolpad/assets/92228082/2138eb52-3a8b-4063-aaf4-ebc34e747c6e">
